PostHog/PostHogStorage.swift

```diff
@@ -10,11 +10,24 @@ import Foundation
 /**
  # Storage
 
- posthog-ios stores data either to file or to UserDefaults in order to support tvOS. As recordings won't work on tvOS anyways and we have no tvOS users so far,
- we are opting to only support iOS via File storage.
+ posthog-ios stores data either to file or to UserDefaults in order to support tvOS.
+
+ Note for tvOS:
+ As tvOS restricts access to persisted Application Support directory, we use Library/Caches instead for storing file queues
+ and UserDefaults for persisting more lightweight data like anonymous ids, feature flags etc.
+
+ According to Apple, you can use UserDefaults to store up to 500KB of data on tvOS
+ see: https://developer.apple.com/forums/thread/16967?answerId=50696022#50696022
  */
 func applicationSupportDirectoryURL() -> URL {
-    let url = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
+    #if os(tvOS)
+        // tvOS restricts access to Application Support directory on physical devices
+        // Use Library/Caches directory which may have less frequent eviction behavior than temp (which is purged when the app quits)
+        let url = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
+    #else
+        let url = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
+    #endif
```

Comment:
we can do similar to `librarySubPath` here
define the param with the if condition, so we don't repeat the whole `FileManager` line